sdks/go/pkg/beam/options/jobopts/options_test.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobopts
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core"
+)
+
+func TestGetEndpoint(t *testing.T) {
+	address := "localhost:8099"
+	Endpoint = &address
+	if v, err := GetEndpoint(); err == nil {
```

Comment:
This is such a small nit, but I think the cleaner way to handle this is

```
v, err := GetEndpoint()
if err != nil {
     t.Fatalf("unexpected error GetEndpoint(): %v", err)
}
if v != address {
     t.Errorf("GetEndpoint() = %v, want %v", v, address)
}
```

It flows a little more and reads easier at a glance to me, especially when that pattern is applied to a more involved test.

Reply:
Make sense.